lib/kubeclient/common.rb

```diff
@@ -484,5 +485,26 @@ def http_options(uri)
 
       options.merge(@socket_options)
     end
+
+    def use_default_gcp_token
+      scopes =  ['https://www.googleapis.com/auth/cloud-platform']
+      authorization = Google::Auth.get_application_default(scopes)
+
+      authorization.apply({})
+
+      authorization.access_token
+    rescue Signet::AuthorizationError => e
```

Comment:
Is this wrapping helping?  It was suggested "for consistency" but perhaps it's nicer to be able to rescue original Signet exceptions rather than a kubeclient exception?
Plus, we rebranded our main exception as `Kubeclient::HttpError` this makes less sense.